app/controllers/concerns/saml_idp_auth_concern.rb

```diff
@@ -46,13 +46,23 @@ def loa2_or_above?
   end
 
   def identity_not_verified?
-    UserDecorator.new(current_user).identity_not_verified?
+    !current_user.verified?
```

Comment:
Sorry if I wasn't clear earlier. When I said +1 to fixing the User model in a separate PR, I meant for the legacy methods. Since the `verified?` method is new to this PR, it should live in the UserDecorator class.